Build error in proj.cpp after commit fde3d58

<!-- Please search existing issues to avoid creating duplicates. -->


#### Example of problem
After commit fde3d58 by @jjimenezshaw, the following build setup (from GDAL):

```shell
export CC="ccache gcc"
export CXX="ccache g++"
export PROJ_DB_CACHE_DIR="$HOME/.ccache"
export CFLAGS="-DPROJ_RENAME_SYMBOLS -O2 -g"
export CXXFLAGS="-DPROJ_RENAME_SYMBOLS -DPROJ_INTERNAL_CPP_NAMESPACE -O2 -g"

cmake . \
      -DBUILD_SHARED_LIBS=ON \
      -DCMAKE_INSTALL_PREFIX=/usr/local \
      -DBUILD_TESTING=OFF
make -j20
```
started giving this error on Ubuntu 22.04 LTS:

```
...
[ 99%] Linking CXX executable ../../bin/geod
[ 99%] Built target geod
[100%] Generating invgeod
[100%] Built target invgeod
[91m/proj/src/apps/proj.cpp: In function 'int main(int, char**)':
/proj/src/apps/proj.cpp:544:54: error: ISO C++ forbids declaration of 'type name' with no type [-fpermissive]
  544 |                     auto crs = dynamic_cast<const crs::ProjectedCRS *>(
      |                                                      ^~
/proj/src/apps/proj.cpp:544:51: error: expected '>' before 'crs'
  544 |                     auto crs = dynamic_cast<const crs::ProjectedCRS *>(
      |                                                   ^~~
/proj/src/apps/proj.cpp:544:51: error: expected '(' before 'crs'
  544 |                     auto crs = dynamic_cast<const crs::ProjectedCRS *>(
      |                                                   ^~~
      |                                                   (
/proj/src/apps/proj.cpp:544:51: error: 'crs' is not a class, namespace, or enumeration
  544 |                     auto crs = dynamic_cast<const crs::ProjectedCRS *>(
      |                                                   ^~~
/proj/src/apps/proj.cpp:544:70: error: expected primary-expression before '>' token
  544 |                     auto crs = dynamic_cast<const crs::ProjectedCRS *>(
      |                                                                      ^
/proj/src/apps/proj.cpp:545:42: error: expected ')' before ';' token
  545 |                         P->iso_obj.get());
      |                                          ^
      |                                          )
/proj/src/apps/proj.cpp:548:42: error: 'cs' has not been declared
  548 |                     swapAxisCrs = dir == cs::AxisDirection::NORTH ||
      |                                          ^~
/proj/src/apps/proj.cpp:549:42: error: 'cs' has not been declared
  549 |                                   dir == cs::AxisDirection::SOUTH;
      |                                          ^~
[0m[91mmake[2]: *** [src/apps/CMakeFiles/binproj.dir/build.make:76: src/apps/CMakeFiles/binproj.dir/proj.cpp.o] Error 1
[0m[91mmake[1]: *** [CMakeFiles/Makefile2:565: src/apps/CMakeFiles/binproj.dir/all] Error 2
```

#### Environment Information
 - PROJ version (`proj`)
HEAD
 - Operation System Information
Ubuntu 22.04 LTS
gcc (Ubuntu 11.3.0-1ubuntu1~22.04.1) 11.3.0

#### Installation method
Cloned repository.

--- a/src/apps/proj.cpp
+++ b/src/apps/proj.cpp
@@ -540,13 +540,12 @@ int main(int argc, char **argv) {
             const auto type = proj_get_type(P);
             if (type == PJ_TYPE_PROJECTED_CRS) {
                 try {
-                    using namespace osgeo::proj;
-                    auto crs = dynamic_cast<const crs::ProjectedCRS *>(
+                    auto crs = dynamic_cast<const NS_PROJ::crs::ProjectedCRS *>(
                         P->iso_obj.get());
                     auto dir =
                         crs->coordinateSystem()->axisList()[0]->direction();
-                    swapAxisCrs = dir == cs::AxisDirection::NORTH ||
-                                  dir == cs::AxisDirection::SOUTH;
+                    swapAxisCrs = dir == NS_PROJ::cs::AxisDirection::NORTH ||
+                                  dir == NS_PROJ::cs::AxisDirection::SOUTH;
                 } catch (...) {
                 }
                 auto ctx = P->ctx;

